pkg/mutator/http_handler.go

```diff
@@ -0,0 +1,205 @@
+package mutator
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/giantswarm/microerror"
+	"k8s.io/api/admission/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	metav1beta1 "k8s.io/apimachinery/pkg/apis/meta/v1beta1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/giantswarm/azure-admission-controller/pkg/filter"
+)
+
+var (
+	scheme        = runtime.NewScheme()
+	codecs        = serializer.NewCodecFactory(scheme)
+	Deserializer  = codecs.UniversalDeserializer()
+	InternalError = errors.New("internal admission controller error")
+)
+
+type HttpHandlerFactoryConfig struct {
+	CtrlClient client.Client
+}
+
+type HttpHandlerFactory struct {
+	ctrlClient client.Client
+}
+
+func NewHttpHandlerFactory(config HttpHandlerFactoryConfig) (*HttpHandlerFactory, error) {
+	if config.CtrlClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.CtrlClient must not be empty", config)
+	}
+
+	h := &HttpHandlerFactory{
+		ctrlClient: config.CtrlClient,
+	}
+
+	return h, nil
+}
+
+func (h *HttpHandlerFactory) NewHttpCreateHandler(mutator WebhookCreateHandler) http.HandlerFunc {
+	mutateFunc := func(ctx context.Context, review v1beta1.AdmissionReview) ([]PatchOperation, error) {
+		object, err := mutator.Decode(review.Request.Object)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+		ok, err := filter.IsCRProcessed(ctx, h.ctrlClient, object.GetObjectMeta())
```

Comment:
Checking if `azure-admission-controller` should mutate CR. If this returns `false`, mutation is not called at all. Same is done for update below.